gcp/src/ARCFileReader.cxx

```diff
@@ -124,24 +130,31 @@ ARCFileReader::ARCFileReader(const std::string &path, double jiffie_base) :
 
 
 
-ARCFileReader::ARCFileReader(const std::vector<std::string> &filename, double jiffie_base) :
-    ms_jiffie_base_(jiffie_base)
+ARCFileReader::ARCFileReader(const std::vector<std::string> &filename,
+                             unsigned experiment) : experiment(experiment)
 {
+	if (experiment == Experiments::SPT || experiment == Experiments::BK) {
+		ms_jiffie_base_ = G3Units::ms;
+	} else if (experiment == Experiments::PB) {
+		ms_jiffie_base_ = 86400/INT_MAX;
+	} else {
+		log_fatal("Unrecognized Experiment");
+	}
+
 	if (filename.size() == 0)
 		log_fatal("Empty file list provided to G3Reader");
 
 	for (auto i = filename.begin(); i != filename.end(); i++){
 		boost::filesystem::path fpath(*i);
 		if (!boost::filesystem::exists(fpath) ||
-		    !boost::filesystem::is_regular_file(fpath)) 
+		    !boost::filesystem::is_regular_file(fpath))
 			log_fatal("Could not find file %s", i->c_str());
 		filename_.push_back(*i);
 	}
 
 	const std::string path = filename_.front();
 	filename_.pop_front();
 	StartFile(path);
-
```

Comment:
Could you strip the extraneous whitespace-only changes from the diff (here, above, and various other places)?